righttyper/righttyper.py

```diff
@@ -243,8 +243,16 @@ def enter_function(code: CodeType, offset: int) -> Any:
             }
         else:
             defaults = {}
-
-        process_function_arguments(t, inspect.getargvalues(frame), defaults)
+        
+        self_type: type | None = None
+        args = inspect.getargvalues(frame).args
```

Comment:
Why invoke `getargvalues` a second time here?  Maybe pass `function` into `process_function_arguments`, moving the default computation there, and then use the `args` it already has in there to figure out `self_type`...